Wallet details update ui when creating new wallet on group

greg-schrammel opened the pull request:
Fixes BX-####
Figma link (if any):

## What changed (plus any additional context for devs)

tbh I'm not sure what `getSettingWallets`/`setSettingWallets` is for, tried to fix it without going much deeper on it lmk if theres something I should be aware of 

## Screen recordings / screenshots

before / after

https://github.com/rainbow-me/browser-extension/assets/6232729/55ffd407-7985-4cde-bd44-5c3d650b3226


https://github.com/rainbow-me/browser-extension/assets/6232729/9d82006c-414c-4573-855c-d3ab74e1e1ea


<!-- Screen recordings can also be helpful for showing reviewers what to test for.  -->

## What to test

<!--

Please be thorough about what to test to help reviewers.
You might want to emphasize potential regressions to check for.
If your code relies on a feature flag for checking both paths of the feature flag, other parts of the code that may have been impacted by your changes, etc.

Don't know what to write here? List all the steps you did to test the changes. This might help QA better understand what/how to test.

-->

derHowie commented:
Can you add a little bit of detail regarding what this fixes? I'm unsure I understand what this change does.

greg-schrammel commented:
> Can you add a little bit of detail regarding what this fixes? I'm unsure I understand what this change does.

note in the before video that I deleted a wallet then created a new one in the same group and clicked back from the show pk screen, the group had only the first wallet and the one I had just deleted and no new one, I then try to delete again and it errors

brunobar79 reviewed (approved):
LGTM!
